Physical_Engine/Query/MaterialComposition.cs

```diff
@@ -195,10 +201,11 @@ private static MaterialComposition MaterialComposition(this Construction prop)
         {
             if (prop.Layers.Any(x => x.Material == null))
             {
-                Engine.Base.Compute.RecordError("The Construction MaterialComposition could not be queried as no Material has been assigned.");
-                return null;
+                Engine.Base.Compute.RecordWarning("At least one Material in a Layered surface property was null. MaterialConstruction excludes this layer, assuming it is void space.");
```

Comment:
After testing, this method still throws an error "1. Cannot evaluate MaterialComposition because the List is empty. "  
With your proposal, you will have 2 messages instead of one to deal with.

Reply:
Hmm, the first error sounds better- this is if you feed an empty list of layers to the Layered SurfaceProperty? If you get the first error, the method should stop and not give you any more.
The second is a warning, as voids will often be included intentionally, but of course sometimes it could be unintentional.

Reply:
I think I need to include a null check before this, so that if none of the layers have a material, it gives up immediately.